x/parallel/parallel.go

```diff
@@ -0,0 +1,62 @@
+// Package parallel implements experimental utilities for performing parallel
+// client interactions with the Authzed API.
+package parallel
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+	"github.com/authzed/authzed-go/internal/ctxgroup"
+)
+
+// Check performs the provided list of CheckRequests in parallel and returns
+// a slice of responses that are stored at their respective indices.
+func Check(ctx context.Context, client api.ACLServiceClient, in []*api.CheckRequest, opts ...grpc.CallOption) ([]*api.CheckResponse, error) {
+	results := make([]*api.CheckResponse, len(in))
+	g := ctxgroup.WithContext(ctx)
+	for i, request := range in {
+		i, request := i, request // close over the value that will not change
```

Comment:
Not sure what you mean. This has to be done so that you don't close over the variables that will mutate each iteration.
This how the official Go team examples all handle this corner-case.

I could explicitly _not_ close over them and pass them as parameters.

Reply:
That's what I meant; I was also saying if we kept to this style, we should rename them, but if this is the official style, then ok